# Fundamentals-of-Data-Analysis - Project
Repository for Fundamentals of Data Analysis module in ATU

Readme by Gerard Ball

## Project brief

+ Create a Jupyter notebook investigating the variables and data points within the well-known iris flower data-set associated with Ronald A Fisher.
 
+ In the notebook, you should discuss the classification of each variable within the data set according to common variable types and scales of measurement in mathematics, statistics and Python. 

+ Select, demonstrate and explain the most appropriate summary statistics to describe each variable. 

+ Select, demonstrate and explain the most appropriate plots for each variable

+ The notebook should follow a cohesive narrative about the data set.

## Project objectives

**Data Loading plus Exploration:**
- Brief outline of the Iris data-set. 
- Load the dataset into a Jupyter notebook in python with sklearn. From here, explore its characteristics to ensure all is well. 



**Iris Variable Classification:**
- Classify each variable in the Iris dataset. 
- Identify variable as type categorical or numerical variables etc
- Further categorize numerical variables as continuous or discrete.


**Scales of Measurement:**
- Outline the scales of measurement for each variable and reasoning.

**Summary Statistics:**
- Complete summary statistics for the dataset.
 -Display the summary statistics.

**Plots for Visualization:**
Choose visualisations like plotting for the variables

**Explanation and Narrative:**
Attempt to construct a clear, concise, cohesive narrative until the end. As such, I hope to make liberal use of markdon explanations throughout to help further explain my logic to the reader. 


**Conclusion:**
Summarize findings and conclude the notebook.



### Data Loading plus Exploration 

The Fisher Iris data set; what is it? 

>"The Iris flower data set or Fisher's Iris data set is a multivariate data set introduced by the British statistician, eugenicist, and biologist Ronald Fisher in his 1936 paper The use of multiple measurements in taxonomic problems as an example of linear discriminant analysis. It is sometimes called Anderson's Iris data set because Edgar Anderson collected the data to quantify the morphologic variation of Iris flowers of three related species. Two of the three species were collected in the Gaspé Peninsula "all from the same pasture, and picked on the same day and measured at the same time by the same person with the same apparatus" - [paperswithcode]https://paperswithcode.com/dataset/iris-1

Put simply; The Iris dataset comprises measurements from three different species of iris flowers - namely, Setosa, Versicolor and Virginica. For these three species, the following variables belong: Sepal Length - the length of the flower's outer petal, Sepal Width - the Width of the flower's outer petal, Petal Length - the length of the flower's inner petal and Petal Width - the width of the flower's inner petal. This incredibly popular dataset is used in the world of data anaytics and machine learning as its simple, clear structure, versatility and beginner friendliness makes it an ideal starting point for beginners - such as myself - to see and visualise how key concepts work. Concept like data exploration, visualisation, benchmarking, predictive analysis etc tio name but a few. 

**In the below code cell, I start by simply loading in the Iris by using sklearn to import it into python. I then make sure to print out all rows of the feature data (measurements of the iris flowers - the species is separate as ) to be sure all is well**

In [1]:
# Needed scikit-learn library to import the data set. Handy 
from sklearn import datasets


iris = datasets.load_iris()

print("Feature Data (All rows):\n", iris.data)

Feature Data (All rows):
 [[5.1 3.5 1.4 0.2]
 [4.9 3.  1.4 0.2]
 [4.7 3.2 1.3 0.2]
 [4.6 3.1 1.5 0.2]
 [5.  3.6 1.4 0.2]
 [5.4 3.9 1.7 0.4]
 [4.6 3.4 1.4 0.3]
 [5.  3.4 1.5 0.2]
 [4.4 2.9 1.4 0.2]
 [4.9 3.1 1.5 0.1]
 [5.4 3.7 1.5 0.2]
 [4.8 3.4 1.6 0.2]
 [4.8 3.  1.4 0.1]
 [4.3 3.  1.1 0.1]
 [5.8 4.  1.2 0.2]
 [5.7 4.4 1.5 0.4]
 [5.4 3.9 1.3 0.4]
 [5.1 3.5 1.4 0.3]
 [5.7 3.8 1.7 0.3]
 [5.1 3.8 1.5 0.3]
 [5.4 3.4 1.7 0.2]
 [5.1 3.7 1.5 0.4]
 [4.6 3.6 1.  0.2]
 [5.1 3.3 1.7 0.5]
 [4.8 3.4 1.9 0.2]
 [5.  3.  1.6 0.2]
 [5.  3.4 1.6 0.4]
 [5.2 3.5 1.5 0.2]
 [5.2 3.4 1.4 0.2]
 [4.7 3.2 1.6 0.2]
 [4.8 3.1 1.6 0.2]
 [5.4 3.4 1.5 0.4]
 [5.2 4.1 1.5 0.1]
 [5.5 4.2 1.4 0.2]
 [4.9 3.1 1.5 0.2]
 [5.  3.2 1.2 0.2]
 [5.5 3.5 1.3 0.2]
 [4.9 3.6 1.4 0.1]
 [4.4 3.  1.3 0.2]
 [5.1 3.4 1.5 0.2]
 [5.  3.5 1.3 0.3]
 [4.5 2.3 1.3 0.3]
 [4.4 3.2 1.3 0.2]
 [5.  3.5 1.6 0.6]
 [5.1 3.8 1.9 0.4]
 [4.8 3.  1.4 0.3]
 [5.1 3.8 1.6 0.2]
 [4.6 3.2 1.4 0.2]
 [5.3 3.7 1.5 0.2]
 [5.  3.3 1.4 0.2]
 [7.  3.2 4.7 1.4]
 [6.4

### Iris Data-set Variable Classifications

1 - **Species** (Iris Setosa, Iris Virginica and Iris Versicolor) - Categorical - Nominal. The species is classified as categorical as it represents different categories in the name of the three species and nominal as the categories don't have any inherent order or ranking 

2 - **Sepal Length, Sepal Width, Petal Length and Petal Width** - Numerical - Continuous. These variables are numerical as they are quantitative, numerical in nature. They're continuous bacause they are all values within a given range. 

### Scales of Measurement

1 - **Species** - Nominal. The species variable is a nominal scale of measurement as it represents distinct categories and is not quantitative. 
2 - **Sepal Length, Sepal Width, Petal Length and Petal Width** - Ratio. The other variables aka the feature data of the Iris would be deemed a ratio scale of measurement. This is so as they all a zero point and can be compared using ratios thus the scale of measurement ratio. They also have a natural order and equal intervals. Put simply - as I am quite simple indeed - the feature data is considered a ratio scale of measurement becasue:

 - It contains a 'true' zero. In this case, this means that zero represents nothing as is true for the feature data as 0 Sepal Length means no Sepal Length. As pointed out by my professor Ian - this is deparate and quite different from a relative zero such as, for example - 0 degrees Celcius. This is a relative 0 as it means the temperature is that of the freezing point of water rather that representing that 0 celcius mrepresents 0 heat, which is not the case. 

 - Has a natural order. This simply means that the values can be arranged, say, smallest to largest etc. The iris data-set has a natural order as the length and width of the petals are able to be measured and compared to deduce one is smaller and thus falls behind a larger variable etc. 

 - They all possess equal interval characteristics. This means that any of the two values on a scale are the same. For the iris, the difference between 5 and 7cm of one variable is the same as 2 and 4 on another. 

 






[Scales of Measurement](https://www.scribbr.com/statistics/levels-of-measurement/)

## Summary Statistics

1 - **Species** Seeing as species is a categorical variable, opting for the count of each class within it is worthwhile to better understand the data. In this case, there are 50 of each of the three flowers, as will be shown below in the following code cell. 

2 - **Sepal Length, Sepal Width, Petal Length and Petal Width** Given that these variables are all numerical, it's best to highlight quantitative findings like central tendency (by way of mean), standard deviation, quartiles and min max values. The mean is useful as it, as mentioned, helps show the central tendency of the data by averaging a value. The standard deviation shows variability, quartiles show distribution and the min and max values show the range of measurements. The above was used as justification to code the results, as shown below. 




In [1]:
import pandas as pd
from sklearn.datasets import load_iris
iris = load_iris()
data = pd.DataFrame(iris.data, columns=iris.feature_names)
data['Species'] = iris.target_names[iris.target]


species_counts = data['Species'].value_counts()

print("Species Counts:")
print(species_counts)

Species Counts:
setosa        50
versicolor    50
virginica     50
Name: Species, dtype: int64


In [4]:
# Summary stats for num vars
numerical_summary = data.describe().loc[['mean', 'std', 'min', '25%', '50%', '75%', 'max']]

print("\nNumerical Variables Summary:")
print(numerical_summary)


Numerical Variables Summary:
      sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
mean           5.843333          3.057333           3.758000          1.199333
std            0.828066          0.435866           1.765298          0.762238
min            4.300000          2.000000           1.000000          0.100000
25%            5.100000          2.800000           1.600000          0.300000
50%            5.800000          3.000000           4.350000          1.300000
75%            6.400000          3.300000           5.100000          1.800000
max            7.900000          4.400000           6.900000          2.500000


General observations of the summary stats prior to var visualisations

 - **Sepal Length** Mean is 5.84 cm or so. The range goes from 4.3 to 7.9, approximately with a standard deviatipon of 0.82 or so. The distribution, as shown by the quartiles, is somewhat symmetrical with a median of 5.8cm which is fairly close to its mean of 5.84. I calculated the median by deducing that the 50th quartile = 50th percentile, which is the median. This is further proven as the maajority os Sepal Length falls between 5.1cm and 6.4 (25th and 75th percentile)

 - **Sepal Width** Mean is 3.06 cm rounded. Range goes from 2 - 4.4cm, which is notably less than Sepal Length. This shows it is standard dev of 0.43cm, approximately half the variability. The distribution here is, like Sepal Length, somewhat of a symmetrical distribution as shown by its mean value of 3.05cm being only slightly greater than its median value of 3cm. 

 **Petal Length** Mean is approx 3.76 cm. Higest standard deviation of all variables at a rounded 1.77cm. This shows with a relatively huge range of 1-6.9cm, the largest range of all the vars, naturally. The distribution here is left skewed as shown by a median of 4.35, much higher than its mean of 3.75. 

 **Petal Width** Mean is 1.20cm rounded. Second lowest variability as shwon by a standard dev of just 0.76cm. The range goes from 0.1-2.5cm. Like Petal Length, the distribution is left skewed as shown by its median of 1.3cm, being noticeably greater than that of its mean of 1.20cm. 

